### PR TITLE
[Refactor] facade userdefaults

### PR DESCRIPTION
**Pull request by mltbnz**
## 📲 What

This PR is intending to cut the Stores into smaller pieces and instead of injecting or directly referencing `UserDefaults` using Protocols to define what properties are accessible from an injected resource.

The `UserDefaults` then conform to these Protocols.

## 🤔 Why

* Clearer dependencies
* Removes UserDefaults from Tests

### CODE REVIEW

File: CriticalMass/ChatInputViewController.swift [hunk 1]
```diff
@@ -23,9 +23,8 @@ final class ChatInputViewController: UIViewController, IBConstructable {
     }
 
     @IBOutlet private var sendButton: UIButton!
-
+    var themeController: ThemeController!
```
Comment (mltbnz) [inline CriticalMass/ChatInputViewController.swift:26]: now gets injected from the AppController

File: CriticalMass/MapViewController.swift [hunk 2]
```diff
@@ -15,33 +15,13 @@ class MapViewController: UIViewController {
     private let nextRideManager: NextRideManager
 
     private let mapInfoViewController = MapInfoViewController.fromNib()
-
```
Comment (mltbnz) [inline CriticalMass/MapViewController.swift:18]: this is just moving a couple lines of code

Comment (fbernutz) [reply CriticalMass/MapViewController.swift:18]: 👍

File: CriticalMassTests/RatingHelperTests.swift [hunk 3]
```diff
@@ -19,71 +19,96 @@ class MockRatingRequest: RatingRequest {
...
+            ratingStorage: ratingsStorage,
             ratingRequest: MockRatingRequest.self,
             currentVersion: "MockVersion"
         )
         execute(times: 5, ratingHelper.onLaunch())
         XCTAssertEqual(MockRatingRequest.requestCounter, 1)
     }
 }
+
+struct RatingStorageMock: RatingStorage {
```
Comment (fbernutz) [inline CriticalMassTests/RatingHelperTests.swift:109]: Maybe it's a good idea to make all of the mock structs in tests private so it's only available in the same file?

**Review by fbernutz — APPROVED** [review]
Looks good for me :)

File: CriticalMassTests/RatingHelperTests.swift (hunk 3 above)
Comment (mltbnz) [reply CriticalMassTests/RatingHelperTests.swift:109]: For some it is actually. Others are needed in more then one place

**Pull request merged**